### DISCUSSION TITLE
Support for Inertia's "first visit" aware lazy loading?

### DISCUSSION THREAD

**Discussion by jameshulse** [spatie/laravel-data#463] [Q&A] [answered]
I have set up laravel-data to work with our Inertia app, but we were making use of Inertia's ability to load data on first load, but then automatically swap to lazy loading for subsequent loads. See their documentation here: https://inertiajs.com/partial-reloads#lazy-data-evaluation

```php
return Inertia::render('Users/Index', [
    // ALWAYS included on first visit...
    // OPTIONALLY included on partial reloads...
    // ALWAYS evaluated...
    'users' => User::get(),

    // ALWAYS included on first visit...
    // OPTIONALLY included on partial reloads...
    // ONLY evaluated when needed...
    'users' => fn () => User::get(),

    // NEVER included on first visit...
    // OPTIONALLY included on partial reloads...
    // ONLY evaluated when needed...
    'users' => Inertia::lazy(fn () => User::get()),
]);
```

Note the second usage in their example which is to pass a callable instead of their `Lazy` object.,

This is particularly useful for loading a somewhat complex page, but then having individual components able to update and pass `only: [...]` to the request to choose which parts of the response to include.

Am I correct in understanding that laravel-data doesn't currently support this?

**Comment by rubenvanassche** [comment, marked as answer]
We have `Lazy::closure` which handles the second example, even use it in Flare.

**Reply by jameshulse** [reply]
That's fantastic! I couldn't find that documented, but it will be very useful for us. Thanks for the reply.
